tools/serverpod_cli/test/generator/dart/server_code_generator/class_field_test.dart

```diff
@@ -112,6 +112,329 @@ void main() {
+              'Missing type alias for "${testClassName}ExpressionBuilder".');
+    });
+  });
+
+  group(
+      'Given a class with table name and persistent field when generating code',
+      () {
+    var entities = [
+      ClassDefinitionBuilder()
+          .withClassName(testClassName)
+          .withFileName(testClassFileName)
+          .withTableName('example_table')
+          .withField(
+            SerializableEntityFieldDefinition(
+                name: 'title',
+                type: TypeDefinition(className: 'String', nullable: true),
+                scope: EntityFieldScopeDefinition.all,
+                shouldPersist: true),
+          )
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    var compilationUnit = parseString(content: codeMap[expectedFilePath]!).unit;
+    var maybeClassNamedExampleTable =
+        CompilationUnitHelpers.tryFindClassDeclaration(compilationUnit,
+            name: '${testClassName}Table');
+
+    group('then class named ${testClassName}Table', () {
+      test('has class variable for field.', () {
+        var exampleTableClass = maybeClassNamedExampleTable!;
+        expect(
+            CompilationUnitHelpers.hasFieldDeclaration(exampleTableClass,
+                name: 'title', isFinal: true),
+            isTrue,
+            reason: 'Missing declaration for title field.');
+      });
+
+      test('has field included in columns.', () {
+        var exampleTableClass = maybeClassNamedExampleTable!;
+        expect(
+            CompilationUnitHelpers.hasMethodDeclaration(exampleTableClass,
+                name: 'columns', functionExpression: '[id, title]'),
+            isTrue,
+            reason: 'Missing title field in columns.');
+      });
+    },
+        skip: maybeClassNamedExampleTable == null
+            ? 'Could not run test because ${testClassName}Table class was not found'
+            : false);
+  });
+
+  group(
+      'Given a class with table name and NON persistent field when generating code',
+      () {
+    var entities = [
+      ClassDefinitionBuilder()
+          .withClassName(testClassName)
+          .withFileName(testClassFileName)
+          .withTableName('example_table')
+          .withField(
+            SerializableEntityFieldDefinition(
+                name: 'title',
+                type: TypeDefinition(className: 'String', nullable: true),
+                scope: EntityFieldScopeDefinition.all,
+                shouldPersist: false),
+          )
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    var compilationUnit = parseString(content: codeMap[expectedFilePath]!).unit;
+    var maybeClassNamedExampleTable =
+        CompilationUnitHelpers.tryFindClassDeclaration(compilationUnit,
+            name: '${testClassName}Table');
```

Comment:
Question: could this be moved into the group instead?

I think it becomes more clear if the last method call here is the actual method we are testing.

Reply:
Hmm, The group we are testing is skipped based on the value of this variable (since the existence of the class creation is tested elsewhere. Therefore, I don't think it makes sense to move it in there. It is a given for the group.